satpy/readers/_geos_area.py

```diff
@@ -41,7 +41,7 @@ def get_xy_from_linecol(line, col, offsets, factors):
     return x__, y__
 
 
-def make_ext(ll_x, ur_x, ll_y, ur_y, h):
+def make_ext(ll_x, ur_x, ll_y, ur_y, h, round=False):
```

Comment:
What about passing the number of decimals? Like 

```python
def make_ext(..., round_decimals=None):
    ...

my_ext = make_ext(..., round_decimals=4)
```

Reply:
Good idea!

Reply:
I'm not a huge fan of this because the rounding is technically in an arbitrary location. Putting the rounding somewhere else results in different results. The generic `round` kwarg was meant to convey "don't worry about it, we'll round where and how we have to".

Reply:
Eh I *wish* it was as simple as getting to the nearest meter, but in tests that's not what is going on. In ABI the pixel resolution is not a nice round metered number, it is something like 1015.9 meters for the 1km data (this is off the top of my head, no idea if that's accurate). For AHI, I'm not sure it follows the same assumptions/design as ABI down to this level of detail. @simonrp84 might know more details, but I know I've had trouble googling it in the past. If we had a theoretical goal for what the extents would be then we could be more specific about numbers being passed in: the pixels should be 1002.15 meters so the left extent should be (num_cols / 2 * -1002.15).

Reply:
So I never did change the keyword argument to take an integer number of decimal places and I don't think we ever decided on a name for that keyword argument that I considered accurate. As mentioned above I'm not rounding to the nearest/closest meter and I'm not necessarily aligning corners although that is kind of the goal. I think there is conflicting hopes for this functionality.

On one side we don't want a generic non-flexible hardcoded rounding that may be sensor/reader specific. On the other side of the argument, we might not want developers to have to think too hard about all this decimal/rounding number "fudging" so a generic "do the rounding" argument would be nice...but maybe we do want developers to have to think about this even if it is hacky.